src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -402,20 +404,15 @@ bool Leader::RlocMatch(uint16_t aFirstRloc16, uint16_t aSecondRloc16, MatchMode
     return matched;
 }
 
-otError Leader::RlocLookup(uint16_t       aRloc16,
```

Comment:
Adding a note about the change here. The diff here is not easy to follow, so I suggest reading the new code only.

Basically here we are removing two larger methods `RlocLookup()` and `IsStableUpdated()` that are no longer needed in the new approach. Instead we have a group of `Validate()` methods  followed by a group of static helper `ContainsMatchingEntry` and then we have `UpdatePrefix/Service()` methods. There is a short comment at the beginning of each method describing what it does.